packages/flutter/lib/src/widgets/scroll_view.dart

```diff
@@ -1257,26 +1257,17 @@ class ListView extends BoxScrollView {
        childrenDelegate = SliverChildBuilderDelegate(
          (BuildContext context, int index) {
            final int itemIndex = index ~/ 2;
-           final Widget? widget;
            if (index.isEven) {
-             widget = itemBuilder(context, itemIndex);
-           } else {
-             widget = separatorBuilder(context, itemIndex);
-             assert(() {
-               if (widget == null) {
```

Comment:
I wonder why this wasn't flagged by the unnecessary_null_comparison check enabled in https://github.com/flutter/flutter/pull/118849.